### PR TITLE
Fix mock experiment service

### PR DESCRIPTION
**Pull request by genlu**
Created a single shared experiment service for test, and fixed existing tests to use it

@dpoeschl @dibarbet @dotnet/roslyn-ide

### CODE REVIEW

File: src/EditorFeatures/TestUtilities/TestExperimentationService.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using System.Composition;
+using Microsoft.CodeAnalysis.Experiments;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Test.Utilities
+{
+    [Shared]
+    [Export(typeof(MockExperimentationService))]
+    [ExportWorkspaceService(typeof(IExperimentationService), WorkspaceKind.Test), PartNotDiscoverable]
+    internal sealed class MockExperimentationService : IExperimentationService
```
Comment (dibarbet) [inline src/EditorFeatures/TestUtilities/TestExperimentationService.cs]: should the file be renamed to MockExperimentationService to match the type?

File: src/EditorFeatures/TestUtilities/TestExperimentationService.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+using System.Collections.Generic;
+using System.Composition;
+using Microsoft.CodeAnalysis.Experiments;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Test.Utilities
+{
+    [Shared]
+    [Export(typeof(MockExperimentationService))]
```
Comment (dibarbet) [inline src/EditorFeatures/TestUtilities/TestExperimentationService.cs]: just curious, why does it need both Export and ExportWorkspaceService?
And will the PartNotDiscoverable only apply to ExportWorkspaceService?

File: src/EditorFeatures/TestUtilities/TestExperimentationService.cs (hunk 1 above)
Comment (genlu) [reply src/EditorFeatures/TestUtilities/TestExperimentationService.cs]: Renamed the type instead. Good catch

File: src/EditorFeatures/TestUtilities/TestExperimentationService.cs (hunk 2 above)
Comment (genlu) [reply src/EditorFeatures/TestUtilities/TestExperimentationService.cs]: > why does it need both Export and ExportWorkspaceService?
I guess the `Export` it's not required, but it saves me a conversion :)

 >  will the PartNotDiscoverable only apply to ExportWorkspaceService
I believe so.

**Comment by genlu** [comment]
@dibarbet Merged. You will need to fix your tests in the codeflow PR tomorrow

**Pull request merged**